node/pkg/transfer-verifier/transfer-verifier-evm-structs.go

```diff
@@ -0,0 +1,989 @@
+package transferverifier
+
+// TODO
+// Change constant naming convention to PascalCase (maybe goimports can do this automatically)
+// Can the actual ethCalls be factored into their own function?
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"math/big"
+	"time"
+
+	connectors "github.com/certusone/wormhole/node/pkg/watchers/evm/connectors"
+	"github.com/certusone/wormhole/node/pkg/watchers/evm/connectors/ethabi"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	ethClient "github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+// Event Signatures
+const (
+	// LogMessagePublished(address indexed sender, uint64 sequence, uint32 nonce, bytes payload, uint8 consistencyLevel);
+	EVENTHASH_WORMHOLE_LOG_MESSAGE_PUBLISHED = "0x6eb224fb001ed210e379b335e35efe88672a8ce935d981a6896b27ffdf52a3b2"
+	// Transfer(address,address,uint256)
+	EVENTHASH_ERC20_TRANSFER = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
+	// Deposit(address,uint256)
+	EVENTHASH_WETH_DEPOSIT = "0xe1fffcc4923d04b559f4d29a8bfc6cda04eb5b0d3c460751c2402c5c5cc9109c"
+)
+
+// Function signatures
+var (
+	// wrappedAsset(uint16 tokenChainId, bytes32 tokenAddress) => 0x1ff1e286
+	TOKEN_BRIDGE_WRAPPED_ASSET_SIGNATURE = []byte("\x1f\xf1\xe2\x86")
+	// isWrappedAsset(address token) => 0x1a2be4da
+	TOKEN_BRIDGE_IS_WRAPPED_ASSET_SIGNATURE = []byte("\x1a\x2b\xe4\xda")
+	// decimals() => 0x313ce567
+	ERC20_DECIMALS_SIGNATURE = []byte("\x31\x3c\xe5\x67")
+	// chainId() => 0x9a8a0592
+	WRAPPED_ERC20_CHAIN_ID_SIGNATURE = []byte("\x9a\x8a\x05\x92")
+)
+
+// Fixed addresses
+var (
+	// https://etherscan.io/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+	WETH_ADDRESS     = common.HexToAddress("c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2")
+	ZERO_ADDRESS     = common.BytesToAddress([]byte{0x00})
+	ZERO_ADDRESS_VAA = VAAAddrFrom(ZERO_ADDRESS)
+)
+
+// EVM chain constants
+const (
+	// The Wormhole Chain ID for the chain being monitored
+	NATIVE_CHAIN_ID = 2
```

Comment:
This makes this Ethereum specific, rather than general EVM. Is that your intent? Maybe you should take the chain ID as an argument? Also, it won't even work for Sepolia / testnet.

Reply:
Good call out! We can make this dynamic.

Reply:
Good suggestion. Made a few changes to accommodate this. Instead of passing it as an argument to the constructor, we can obtain the chain ID from the connector.